src/Aspire.Hosting.Azure/Utils/ResourceGroupNameHelpers.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using System.Text;
+
+namespace Aspire.Hosting.Azure.Utils;
+
+internal static class ResourceGroupNameHelpers
+{
+    public static int MaxResourceGroupNameLength = 90;
+
+    /// <summary>
+    /// Converts or excludes any characters which are not valid resource group name components.
+    /// </summary>
+    /// <param name="resourceGroupName">The text to normalize.</param>
+    /// <returns>The normalized resource group name or an empty string if no characters were valid.</returns>
```

Comment:
does it matter that it could return empty or very short? then it'd be `"{prefix}--{suffix}"

Reply:
I thought about it and decided to not handle that because it's still a valid, unique RG name. The app name won't be part of it, that is all. And if it mattered to users they could still use the new option to provide a custom part, or the existing one to provide the whole.

Reply:
I suppose instead of zero length it could use a hash. but I have no opinion, feel free to leave as is of course.